apps/openmw/mwmechanics/aiwander.cpp

```diff
@@ -303,8 +334,55 @@ namespace MWMechanics
 
             if (storage.mPathFinder.isPathConstructed())
             {
-                storage.mState = Wander_Walking;
+                storage.setState(Wander_Walking);
+            }
+        }
+    }
+
+    /*
+     * Commands actor to walk to a random location near original spawn location.
+     *
+     * Creatures simply wander a certain distance from their starting location, while NPCs wander a scripted
+     * distance (mDistance) from the position where they started the wander package.
+     * http://www.uesp.net/wiki/Tes3Mod:AIWander
+     */
+    void AiWander::wanderNearStart(const MWWorld::Ptr &actor, AiWanderStorage &storage, int wanderDistance, bool isNpc) {
+        const ESM::Pathgrid::Point currentPosition = actor.getRefData().getPosition().pos;
+
+        // Determine a random location within radius of original position
+        const float pi = 3.14159265359f;
+        const float maxWanderDistance = isNpc ? wanderDistance : MINIMUM_WANDER_DISTANCE * 14.0f;
```

Comment:
Currently I'm trying to find out why mDistance is always 0 and where the AIWander Range is getting set (I'm assuming mDistance, but either I'm wrong or there is a bug I need to fix).